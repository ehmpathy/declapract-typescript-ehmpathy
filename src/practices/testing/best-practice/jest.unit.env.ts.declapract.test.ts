import { contents } from './jest.unit.env.ts.declapract';

describe('jest.unit.env.ts', () => {
  it('should include config mock when project uses config practice', async () => {
    const declaredBestPracticeContents = await contents({
      projectPractices: ['config'],
    } as any);
    expect(declaredBestPracticeContents).toContain(
      "jest.mock('./src/utils/config/getConfig', ",
    );
    expect(declaredBestPracticeContents).toMatchSnapshot();
  });
  it('should not include config mock when project does not use config practice', async () => {
    const declaredBestPracticeContents = await contents({
      projectPractices: [],
    } as any);
    expect(declaredBestPracticeContents).not.toContain(
      "jest.mock('./src/utils/config/getConfig', ",
    );
    expect(declaredBestPracticeContents).toMatchSnapshot();
  });
});
