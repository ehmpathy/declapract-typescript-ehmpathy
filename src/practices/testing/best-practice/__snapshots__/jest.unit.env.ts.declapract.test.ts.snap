// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest.unit.env.ts should include config mock when project uses config practice 1`] = `
"import { stage, Stage } from './src/utils/environment';

jest.mock('./src/utils/config/getConfig', () => ({
  getConfig: jest.fn().mockImplementation(() => require('./config/test.json')), // mock that getConfig just returns plaintext test env config in unit tests
}));

/**
 * sanity check that unit tests are only run in 'test' environment
 * - if they are run in prod environment, we could load a bunch of junk data into our prod databases, which would be no bueno
 */
if (stage !== Stage.TEST && process.env.I_KNOW_WHAT_IM_DOING !== 'true')
  throw new Error(\`unit-test is not targeting stage 'test'\`);
"
`;

exports[`jest.unit.env.ts should include not config mock when project does not use config practice 1`] = `
"import { stage, Stage } from './src/utils/environment';

/**
 * sanity check that unit tests are only run in 'test' environment
 * - if they are run in prod environment, we could load a bunch of junk data into our prod databases, which would be no bueno
 */
if (stage !== Stage.TEST && process.env.I_KNOW_WHAT_IM_DOING !== 'true')
  throw new Error(\`unit-test is not targeting stage 'test'\`);
"
`;
